web/api/query.go

```diff
@@ -46,9 +47,9 @@ func httpJSONError(w http.ResponseWriter, err error, code int) {
 	fmt.Fprintln(w, ast.ErrorToJSON(err))
 }
 
-func parseTimestampOrNow(t string) (clientmodel.Timestamp, error) {
+func parseTimestampOrNow(t string, nower utility.Time) (clientmodel.Timestamp, error) {
```

Comment:
It has the benefit that the type's [zero value](https://golang.org/ref/spec#The_zero_value) is a [valid `utility.Time` returning the current time](https://github.com/prometheus/prometheus/blob/master/utility/time.go#L40-L42), meaning that it can be omitted completely (i.e. when you're not in tests) in places where it's used to configure objects like the `MetricsService`. Whether that's worth the trouble is a good question. I don't feel strongly either way.